File: Node/Prisma_ORM/src/bll/services/productService.js
const ProductRepository = require('../../dal/repositories/productRepository');

const ProductService = {
  getAll: async () => {
    return await ProductRepository.getAll();
  },

  getById: async (id) => {
    const entity = await ProductRepository.getById(id);
    if (!entity) throw new Error('entity not found');
    return entity;
  },

  create: async (entity) => {
    return await ProductRepository.create(entity);
  },

  update: async (id, entity) => {
    return await ProductRepository.update(id, entity);
  },

  delete: async (id) => {
    return await ProductRepository.delete(id);
  }
};

module.exports = ProductService; 

File: Go/GORM_ORM/bll/services/productService.go
package services

import (
	"GORM_ORM/dal/entities"
	"GORM_ORM/dal/repositories"
)

type ProductService interface {
	GetAll() ([]entities.Product, error)
	GetByID(id int) (*entities.Product, error)
	Create(product *entities.Product) error
	Update(product *entities.Product) error
	Delete(id int) error
}

type productService struct {
	repo repositories.ProductRepository
}

func NewProductService(repo repositories.ProductRepository) ProductService {
	return &productService{repo: repo}
}

func (s *productService) GetAll() ([]entities.Product, error) {
	return s.repo.GetAll()
}

func (s *productService) GetByID(id int) (*entities.Product, error) {
	return s.repo.GetByID(id)
}

func (s *productService) Create(product *entities.Product) error {
	return s.repo.Create(product)
}

func (s *productService) Update(product *entities.Product) error {
	return s.repo.Update(product)
}

func (s *productService) Delete(id int) error {
	return s.repo.Delete(id)
}
